advanced/wallets/react-wallet-v2/src/views/SessionProposalModal.tsx

```diff
@@ -254,8 +254,8 @@ export default function SessionProposalModal() {
       setIsLoadingApprove(true)
       try {
         if (reorderedEip155Accounts.length > 0) {
-          // reorderedEip155Accounts includes Smart Accounts(if enabled) and EOA's
-          namespaces.eip155.accounts = reorderedEip155Accounts
+          // we should append the smart accounts to the available eip155 accounts
+          namespaces.eip155.accounts = reorderedEip155Accounts.concat(namespaces.eip155.accounts)
```

Comment:
```suggestion
          namespaces.eip155.accounts = [...new Set(reorderedEip155Accounts.concat(namespaces.eip155.accounts))]
```
this will remove any duplicates from the array

Reply:
Fixed this by modifying the supportedAddressPriority method to return only the smart account address on supported chain for enabled SA. Will not need this as now no duplication will occur.